packages/flutter/lib/src/widgets/heroes.dart

```diff
@@ -304,6 +304,11 @@ class Hero extends StatefulWidget {
             inviteHero(hero, tag);
           }
         }
+      } else if (element.widget is Offstage) {
+        final Offstage offstageWidget = element.widget;
+        if (offstageWidget.offstage) {
```

Comment:
You can omit final declaration if your condition looks like this:
```
if ((element.widget as Offstage).offstage) {
```

Reply:
I think it's preferable to write in a simple way that anyone read and understand immediately.

At first I thought to write as follows:

```dart
final Widget widget = element.widget;
if (widget is Hero) {
  ...
} else if (widget is Offstage) {
  if (widget.offstage) {
    return;
  }
}
```

Reply:
The current implementation is the same as the implementation in the if statement of Hero.